File: wallet-android/chapter9/code/app/src/main/java/com/digquant/util/ThreadPoolUtil.java
package com.digquant.util;

import java.util.concurrent.LinkedBlockingDeque;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

public class ThreadPoolUtil {
    private static ThreadPoolExecutor executor = new ThreadPoolExecutor(
            2,
            10,
            10 * 60,
            TimeUnit.SECONDS,
            new LinkedBlockingDeque<>());

    public static void AddTask(Runnable runnable) {
        executor.execute(runnable);
    }
}

File: wallet-android/chapter1/code/app/src/main/java/com/digquant/util/ThreadPoolUtil.kt
package com.digquant.util

import java.util.concurrent.LinkedBlockingDeque
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.TimeUnit

class ThreadPoolUtil {
    companion object {

        private val executor = ThreadPoolExecutor(
            2,
            10,
            10 * 60,
            TimeUnit.SECONDS,
            LinkedBlockingDeque()
        )

        fun AddTask(runnable: Runnable?) {
            executor.execute(runnable)
        }
    }
}